src/stats/violin.jl

```diff
@@ -32,32 +35,60 @@ end
 conversion_trait(x::Type{<:Violin}) = SampleBased()
 
 function plot!(plot::Violin)
-    x, y, width, side, show_median = plot[1], plot[2], plot[:width], plot[:side], plot[:show_median]
-    npoints, boundary, bandwidth = plot[:npoints], plot[:boundary], plot[:bandwidth]
-    dodge, n_dodge, x_gap, dodge_gap = plot[:dodge], plot[:n_dodge], plot[:x_gap], plot[:dodge_gap]
-
-    signals = lift(x, y, width, dodge, n_dodge, x_gap, dodge_gap, side, show_median, npoints, boundary, bandwidth) do x, y, width, dodge, n_dodge, x_gap, dodge_gap, vside, show_median, n, bound, bw
+    x, y = plot[1], plot[2]
+    args = @extract plot (width, side, color, show_median, npoints, boundary, bandwidth,
+        datalimits, max_density, dodge, n_dodge, x_gap, dodge_gap)
+    signals = lift(x, y, args...) do x, y, width, side, color, show_median, n, bound, bw, limits, max_density, dodge, n_dodge, x_gap, dodge_gap
         x̂, violinwidth = xw_from_dodge(x, width, 1, x_gap, dodge, n_dodge, dodge_gap)
-        vertices = Vector{Point2f0}[]
-        lines = Pair{Point2f0, Point2f0}[]
-        for (key, idxs) in StructArrays.finduniquesorted(x̂)
+
+        # Allow `side` and `color` to be either scalar or vector
+        sides = broadcast(x̂, side) do _, s
+            return s == :left ? - 1 : s == :right ? 1 : 0
+        end
+        colors = broadcast(x̂, color) do _, c
+            rgba = to_color(c)
+            # `RGBA` values cannot be sorted
+            return (rgba.r, rgba.g, rgba.b, rgba.alpha)
+        end
+
+        sa = StructArray((x = x̂, side = sides, color = colors))
```

Comment:
Not sure if one should also group by `color` here, it is a bit of an odd interface to have to pass a vector of colors that is the same length as a vector of sides. Maybe there could be attributes `leftcolor` and `rightcolor` instead?